src/main/io/gps.c

```diff
@@ -2283,38 +2291,25 @@ static bool UBLOX_parse_gps(void)
         break;
     case CLSMSG(CLASS_CFG, MSG_CFG_GNSS):
         {
-            bool isSBASenabled = false;
-            bool isM8NwithDefaultConfig = false;
-
-            if ((ubxRcvMsgPayload.ubxCfgGnss.numConfigBlocks >= 2) &&
-                (ubxRcvMsgPayload.ubxCfgGnss.configblocks[1].gnssId == 1) && //SBAS
-                (ubxRcvMsgPayload.ubxCfgGnss.configblocks[1].flags & UBLOX_GNSS_ENABLE)) { //enabled
-
-                isSBASenabled = true;
-            }
-
-            if ((ubxRcvMsgPayload.ubxCfgGnss.numTrkChHw == 32) &&  //M8N
-                (ubxRcvMsgPayload.ubxCfgGnss.numTrkChUse == 32) &&
-                (ubxRcvMsgPayload.ubxCfgGnss.numConfigBlocks == 7) &&
-                (ubxRcvMsgPayload.ubxCfgGnss.configblocks[2].gnssId == 2) && //Galileo
-                (ubxRcvMsgPayload.ubxCfgGnss.configblocks[2].resTrkCh == 4) && //min channels
-                (ubxRcvMsgPayload.ubxCfgGnss.configblocks[2].maxTrkCh == 8) && //max channels
-                !(ubxRcvMsgPayload.ubxCfgGnss.configblocks[2].flags & UBLOX_GNSS_ENABLE)) { //disabled
-
-                isM8NwithDefaultConfig = true;
-            }
-
             const uint16_t messageSize = 4 + (ubxRcvMsgPayload.ubxCfgGnss.numConfigBlocks * sizeof(ubxConfigblock_t));
-
             ubxMessage_t tx_buffer;
+
             memcpy(&tx_buffer.payload, &ubxRcvMsgPayload, messageSize);
```

Comment:
```suggestion
            memcpy(&tx_buffer.payload, &ubxRcvMsgPayload, MIN(messageSize, sizeof(tx_buffer.payload)));
```
Just to formally prevent buffer overflow on invalid numConfigBlocks

Reply:
Result:

 ../main/io/gps.c:2297:21: error: taking address of packed member 'payload' of class or structure 'ubxMessage_s' may result in an unaligned pointer value [-Werror,-Waddress-of-packed-member]
            memcpy(&tx_buffer.payload, &ubxRcvMsgPayload, MIN(messageSize, sizeof(tx_buffer.payload)));
                    ^~~~~~~~~~~~~~~~~
1 error generated.